R/download_data_wrds_compustat.R

```diff
@@ -85,9 +89,44 @@ download_data_wrds_compustat <- function(type, start_date, end_date, additional_
         inv = at / at_lag - 1,
         inv = if_else(at_lag <= 0, NA, inv)
       )
+  }
+
+  if (grepl("compustat_quarterly", type, fixed = TRUE)) {
+    fundq_db <- tbl(con, in_schema("comp", "fundq"))
+
+    compustat <- fundq_db |>
+      filter(
+        indfmt == "INDL" &
+          datafmt == "STD" &
+          consol == "C" &
+          between(datadate, start_date, end_date)
+      ) |>
+      select(
+        gvkey, datadate, rdq, fqtr, fyearq,
+        all_of(additional_columns)
+      ) |>
+      collect()
 
     disconnection_connection(con)
 
-    processed_data
+    compustat <- compustat |>
+      drop_na(fqtr)|>
+      mutate(date = ceiling_date(datadate, "quarter") %m-% months(1),
+             timepoint = paste0(fyearq, fqtr)) |>
```

Comment:
I'm not convinced that "timepoint" is a clear naming and what its purpose is. Why not just keep two integers "year" and "quarter"? Then users can do whatever they want with it.

Reply:
Removed timepoint. We need it to ensure that only one observation per quarter exists.